hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/PartitionedMapTableResolver.java

```diff
@@ -101,103 +101,129 @@ public PartitionedMapTableResolver(NodeEngine nodeEngine) {
         return res;
     }
 
-    // TODO: VO: Abstract out Jet stuff in a clean way.
-    @SuppressWarnings({"rawtypes", "checkstyle:MethodLength", "checkstyle:CyclomaticComplexity", "checkstyle:NPathComplexity"})
+    /**
+     * @param explicitRequest True, if the table was requested explicitly
+     *     through DDL. In this case the result is non-null and the returned
+     *     table never contains an exception. If false, might return null or a
+     *     table with an exception.
+     */
+    @Nullable
+    @SuppressWarnings({"checkstyle:MethodLength", "checkstyle:CyclomaticComplexity", "checkstyle:NPathComplexity"})
     public static PartitionedMapTable createTable(
```

Comment:
Previously I reverted all changes from this class related to external tables for a reason - to keep a single responsibility principle in place. 

The goal of `PartitionedMapTableResolver` is to expose existing maps through a predefined schema. We should not add external fields handling here. The same is applicable to `ReplicatedMapTableResolver`.

If there are some static methods that are common for both internal and external resolvers, let's factor them out, instead of mixing unrelated concepts inside a single class.